### ISSUE TITLE
`trtllm_fp8_per_tensor_scale_moe` and `trtllm_fp8_block_scale_moe` do not support routing_logits==float32 except for DeepSeek

### ISSUE THREAD

**Issue by mgoin** [flashinfer-ai/flashinfer#2469]
https://github.com/flashinfer-ai/flashinfer/blob/3fa8905c79fe3b711ff62af5bab5e3cc5ec6d5ea/csrc/trtllm_fused_moe_kernel_launcher.cu#L1564-L1570

https://github.com/flashinfer-ai/flashinfer/blob/3fa8905c79fe3b711ff62af5bab5e3cc5ec6d5ea/csrc/trtllm_fused_moe_kernel_launcher.cu#L1664-L1672

This causes issues in vLLM for models like `MiniMaxAI/MiniMax-M2.1` that use `RenormalizeNaive` routing method but require routing logits to be kept at float32, see https://github.com/vllm-project/vllm/issues/33543

This is not an issue for `trtllm_bf16_moe`, `trtllm_fp4_block_scale_moe`, or `trtllm_mxint4_block_scale_moe` as they support float32 or bfloat16 for all cases.

**Comment by pavanimajety** [comment]
@ChristinaZ for visibility.

**Comment by aleozlx** [comment]
@ChristinaZ do you know if there are limitations from the kernel or it's likely just integration issue? thx

**Comment by ChristinaZ** [comment]
I think we can enable this feature by adding some additional (template) instantiations on our side. Let me take a look.

**Comment by yweng0828** [comment]
Hi @mgoin @pavanimajety @aleozlx, under @ChristinaZ 's guidance, we have filed a [PR2534](https://github.com/flashinfer-ai/flashinfer/pull/2534) to remove the restrictions here. Could you please help us to review it? Thank you.
